gvsbuild/projects/expat.py

```diff
@@ -26,6 +26,7 @@ def __init__(self):
             self,
             "expat",
             version="2.5.0",
+            repository="libexpat",
```

Comment:
it is not clear to me why some `repository` are not URLs and what happens in that case

Reply:
repository is a string passed to `lastversion` to identify the repository. It can be the repository URL, or if you're lazy, a name which is processed by `lastversion`.

If repository is not defined, we pass the name. In this case expat is not associated with the repository, so we need to feed `lastversion` with a better identifier:
```
$ lastversion --verbose expat
lastversion.lastversion - INFO - Verbose 1 level output.
lastversion.ProjectHolder - INFO - Using cache directory: C:\Users\sabatin\AppData\Local\lastversion\lastversion\Cache.
lastversion.ProjectHolder - INFO - Created instance of GitHubRepoSession
lastversion.GitHubRepoSession - INFO - Using API token.
lastversion.GitHubRepoSession - INFO - Checking existence of expat/expat
lastversion.GitHubRepoSession - INFO - Got HTTP status code 404 from https://github.com/expat/expat/releases.atom
lastversion.GitHubRepoSession - INFO - Found expat in repo short name cache
lastversion.GitHubRepoSession - INFO - Using repo None obtained from search API
lastversion.GitHubRepoSession - INFO - Got HTTP status code 200 from https://github.com/vic/expat/releases.atom
lastversion.GitHubRepoSession - INFO - Checking tag v1.0.5
lastversion.ProjectHolder - INFO - Sanitizing string v1.0.5 as a satisfying version.
lastversion.ProjectHolder - INFO - Parsed as Version OK. String representation: 1.0.5.
lastversion.GitHubRepoSession - INFO - Got HTTP status code 200 from https://api.github.com/repos/vic/expat/releases
[...]
lastversion.ProjectHolder - INFO - Sanitizing string v1.0.5 as a satisfying version.
lastversion.ProjectHolder - INFO - Parsed as Version OK. String representation: 1.0.5.
lastversion.GitHubRepoSession - INFO - Using commit date
lastversion.ProjectHolder - INFO - Sanitizing string v1.0.4 as a satisfying version.
lastversion.ProjectHolder - INFO - Parsed as Version OK. String representation: 1.0.4.
lastversion.GitHubRepoSession - INFO - Using commit date
lastversion.ProjectHolder - INFO - Sanitizing string v1.0.3 as a satisfying version.
lastversion.ProjectHolder - INFO - Parsed as Version OK. String representation: 1.0.3.
lastversion.GitHubRepoSession - INFO - Using commit date
lastversion.lastversion - INFO - Located the latest release tag v1.0.5 at: https://github.com/vic/expat via feed mechanism
1.0.5
```

When using `libexpat`, the correct repository is identified:
```
$ lastversion --verbose libexpat
lastversion.lastversion - INFO - Verbose 1 level output.
lastversion.ProjectHolder - INFO - Using cache directory: C:\Users\sabatin\AppData\Local\lastversion\lastversion\Cache.
lastversion.ProjectHolder - INFO - Created instance of GitHubRepoSession
lastversion.GitHubRepoSession - INFO - Using API token.
lastversion.GitHubRepoSession - INFO - Checking existence of libexpat/libexpat
lastversion.GitHubRepoSession - INFO - Got HTTP status code 404 from https://github.com/libexpat/libexpat/releases.atom
lastversion.GitHubRepoSession - INFO - Found libexpat in repo short name cache
lastversion.GitHubRepoSession - INFO - Using repo None obtained from search API
lastversion.GitHubRepoSession - INFO - Got HTTP status code 200 from https://github.com/libexpat/libexpat/releases.atom
lastversion.GitHubRepoSession - INFO - Checking tag R_2_5_0
lastversion.ProjectHolder - INFO - Sanitizing string R_2_5_0 as a satisfying version.
lastversion.ProjectHolder - INFO - Parsed as Version OK. String representation: 2.5.0.
lastversion.GitHubRepoSession - INFO - Got HTTP status code 200 from https://api.github.com/repos/libexpat/libexpat/releases
lastversion.GitHubRepoSession - INFO - Selected version as current selection: 2.5.0.
lastversion.GitHubRepoSession - INFO - Checking tag R_2_4_9
lastversion.ProjectHolder - INFO - Sanitizing string R_2_4_9 as a satisfying version.
lastversion.ProjectHolder - INFO - Parsed as Version OK. String representation: 2.4.9.
lastversion.GitHubRepoSession - INFO - Tag R_2_4_9 does not contain newer version than we already found
lastversion.GitHubRepoSession - INFO - Checking tag R_2_4_8
lastversion.ProjectHolder - INFO - Sanitizing string R_2_4_8 as a satisfying version.
lastversion.ProjectHolder - INFO - Parsed as Version OK. String representation: 2.4.8.
lastversion.GitHubRepoSession - INFO - Tag R_2_4_8 does not contain newer version than we already found
lastversion.GitHubRepoSession - INFO - Checking tag R_2_4_7
lastversion.ProjectHolder - INFO - Sanitizing string R_2_4_7 as a satisfying version.
lastversion.ProjectHolder - INFO - Parsed as Version OK. String representation: 2.4.7.
lastversion.GitHubRepoSession - INFO - Tag R_2_4_7 does not contain newer version than we already found
lastversion.GitHubRepoSession - INFO - Checking tag R_2_4_6
lastversion.ProjectHolder - INFO - Sanitizing string R_2_4_6 as a satisfying version.
lastversion.ProjectHolder - INFO - Parsed as Version OK. String representation: 2.4.6.
lastversion.GitHubRepoSession - INFO - Tag R_2_4_6 does not contain newer version than we already found
lastversion.GitHubRepoSession - INFO - Checking tag R_2_4_5
lastversion.ProjectHolder - INFO - Sanitizing string R_2_4_5 as a satisfying version.
lastversion.ProjectHolder - INFO - Parsed as Version OK. String representation: 2.4.5.
lastversion.GitHubRepoSession - INFO - Tag R_2_4_5 does not contain newer version than we already found
lastversion.GitHubRepoSession - INFO - Checking tag R_2_4_4
lastversion.ProjectHolder - INFO - Sanitizing string R_2_4_4 as a satisfying version.
lastversion.ProjectHolder - INFO - Parsed as Version OK. String representation: 2.4.4.
lastversion.GitHubRepoSession - INFO - Tag R_2_4_4 does not contain newer version than we already found
lastversion.GitHubRepoSession - INFO - Checking tag R_2_4_3
lastversion.ProjectHolder - INFO - Sanitizing string R_2_4_3 as a satisfying version.
lastversion.ProjectHolder - INFO - Parsed as Version OK. String representation: 2.4.3.
lastversion.GitHubRepoSession - INFO - Tag R_2_4_3 does not contain newer version than we already found
lastversion.GitHubRepoSession - INFO - Checking tag R_2_4_2
lastversion.ProjectHolder - INFO - Sanitizing string R_2_4_2 as a satisfying version.
lastversion.ProjectHolder - INFO - Parsed as Version OK. String representation: 2.4.2.
lastversion.GitHubRepoSession - INFO - Tag R_2_4_2 does not contain newer version than we already found
lastversion.GitHubRepoSession - INFO - Checking tag R_2_4_1
lastversion.ProjectHolder - INFO - Sanitizing string R_2_4_1 as a satisfying version.
lastversion.ProjectHolder - INFO - Parsed as Version OK. String representation: 2.4.1.
lastversion.GitHubRepoSession - INFO - Tag R_2_4_1 does not contain newer version than we already found
lastversion.lastversion - INFO - Located the latest release tag R_2_5_0 at: https://github.com/libexpat/libexpat via release mechanism
2.5.0
```

Reply:
Thanks, that makes sense